src/lxc/conf.c

```diff
@@ -757,79 +788,80 @@ static int lxc_mount_auto_mounts(struct lxc_conf *conf, int flags, struct lxc_ha
 	}
 
 	if (flags & LXC_AUTO_SHMOUNTS_MASK) {
-		char *new_mount;
-		size_t len_mount;
+		size_t len_cont;
+		char *full_cont_path;
+		struct stat sb;
```

Comment:
Move the declaration of `struct stat sb` into the `if (is_shared_...)` condition since you're only using it there. Keep the scope tight. :)